kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/ReadMetadataRunner.java

```diff
@@ -0,0 +1,122 @@
+import io.delta.kernel.Table;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.defaults.engine.DefaultEngine;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.hadoop.conf.Configuration;
+import org.openjdk.jmh.infra.Blackhole;
+
+public class ReadMetadataRunner implements WorkloadRunner {
+  private final Engine engine;
+  private final Scan scan;
+
+  /**
+   * Constructs the ReadMetadataRunner from the workload spec and the base workload directory
+   * containing test tables. This initializes any state necessary to execute this workload as a
+   * benchmark.
+   *
+   * @param baseWorkloadDirPath The base directory containing workload tables.
+   * @param workload The read_metadata workload specification.
+   */
+  public ReadMetadataRunner(Path baseWorkloadDirPath, ReadMetadata workload) {
+    String resolvedTableRoot = getResolvedTableRoot(baseWorkloadDirPath, workload.getTableRoot());
+    engine = DefaultEngine.create(new Configuration());
+    scan = getScanFromWorkloadSpec(engine, workload, resolvedTableRoot);
+  }
+
+  /**
+   * Resolves the table root path based on whether it's a URI or a relative path. If it is a
+   * relative path, it is resolved based on the workload base directory.
+   *
+   * @param baseWorkloadDirPath The base directory containing workload tables.
+   * @param workloadTableRoot The table root path from the workload spec.
+   * @return The resolved table root path.
+   */
+  private static String getResolvedTableRoot(Path baseWorkloadDirPath, String workloadTableRoot) {
+    String resolvedTableRoot = workloadTableRoot;
+    // If this is not a URI, treat it as a relative path under the base workload directory
+    if (!workloadTableRoot.contains("://")) {
+      // This uses a hard-coded base directory for workload tables. In the future, this could be
+      // made configurable.
+      resolvedTableRoot = baseWorkloadDirPath.resolve(Paths.get(workloadTableRoot)).toString();
+    }
+    return resolvedTableRoot;
+  }
+
+  /**
+   * Constructs the Scan from the workload specification for this runner.
+   *
+   * @param engine The kernel engine to use for table operations.
+   * @param workload The read_metadata workload specification.
+   * @param resolvedTableRoot The resolved table root path.
+   * @return The Scan object used to execute the workload.
+   */
+  private static Scan getScanFromWorkloadSpec(
+      Engine engine, ReadMetadata workload, String resolvedTableRoot) {
+    Table table = Table.forPath(engine, resolvedTableRoot);
+    // Get snapshot (use specified version or latest)
+    Snapshot snapshot;
+    if (workload.getSnapshotVersion() != null) {
+      snapshot = table.getSnapshotAsOfVersion(engine, workload.getSnapshotVersion());
+    } else {
+      snapshot = table.getLatestSnapshot(engine);
+    }
+    // Build scan
```

Comment:
nit: superfluous comment --> this doesn't give us any more context than the line of code below `ScanBuilder scanBuilder = snapshot.getScanBuilder();` :) we can delete this